src/main/kotlin/com/demonwav/mcdev/i18n/intentions/ConvertToTranslationIntention.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.intentions
+
+import com.demonwav.mcdev.i18n.I18nElementFactory
+import com.demonwav.mcdev.util.runWriteAction
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.lang.java.JavaLanguage
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.ui.InputValidatorEx
+import com.intellij.openapi.ui.Messages
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiLiteral
+import com.intellij.psi.codeStyle.JavaCodeStyleManager
+import com.intellij.util.IncorrectOperationException
+
+class ConvertToTranslationIntention : PsiElementBaseIntentionAction() {
+    @Throws(IncorrectOperationException::class)
+    override fun invoke(project: Project, editor: Editor, element: PsiElement) {
+        if (element.parent is PsiLiteral) {
+            val value = (element.parent as PsiLiteral).value as String?
+            val result = Messages.showInputDialogWithCheckBox("Enter translation key:",
+                "Convert String Literal to Translation",
+                "Replace literal with call to I18n (only works on clients!)",
+                true,
+                true,
+                Messages.getQuestionIcon(),
+                null,
+                object : InputValidatorEx {
+                    override fun getErrorText(inputString: String): String? {
+                        if (inputString.isEmpty())
+                            return "Key must not be empty"
+                        if (inputString.contains('='))
+                            return "Key must not contain separator character ('=')"
+                        return null
+                    }
+
+                    override fun checkInput(inputString: String): Boolean {
+                        return !inputString.isEmpty() && !inputString.contains('=')
+                    }
+
+                    override fun canClose(inputString: String): Boolean {
+                        return !inputString.isEmpty() && !inputString.contains('=')
+                    }
+                })
+            if (result.getFirst() != null) {
```

Comment:
I prefer (in general) when using `Pair`, to destructure them as quickly as reasonable (which in this case would be here). This isn't a Kotlin `Pair`, so we can't destructure them in the initial declaration, so just store them locally afterwards.